Versatile Stanza/Spacy SBD and various fixes

lecoqnicolas opened the pull request:
Sorry for the many commits... 
This PR introduces a New Sentencizer Class for StanzaSentencizer. 
The SBD model path is defined as a property of the translation package: it can be stanza, a packaged spacy model (language-specific or not), or take the None value.
The SpacySentencizerSmall class has been edited to support both packaged and non-packaged Spacy models, and also to download and cache correctly the generic "xx_sent_ud_sm" in case the previous property is None (no package). After dowloading once, it loads spacy from the cache instead of downloading on and on.
The Translation classes' property "sentencizer" is defined as of constructing the PackagedTranslation class, using an if branch that takes all SBD model paths and assigns the corresponding sentencizers' classes.
Then, the PR also includes a fix in the tokenizer to allow both byte-fallback tokens and rewriting active underscores.
Meanwhile, some typo corrections, and some reverted bugs that were introduced upon follwing IDE advice.
The PR works in CLI and is tested on all three types of SBD packages.

dhoesele commented on argostranslate/networking.py:
```diff
@@ -76,3 +82,21 @@ def get_from(urls: list[str], retry_count: int = 3) -> bytes | None:
+    info("Downloading spacy model")
+    spacy_model = Path(spacy_cache / "senter" / "model")
+    if not spacy_model.exists():
+        while True:
```
was the while True here intended to retry download in case of failure? Currently it seems to exit on first attempt.

lecoqnicolas replied:
The while True loop makes the function wait for the download to end, or retries in a case of failure.
Let me recheck the loop: I did an edit directly on the machine library for this function, maybe I didn't report it on the PyCharm project.

lecoqnicolas replied:
Can you check what's in your $HOME/.local/cache/argos-translate/spacy directory?
If the loop doesn't work it is empty, if it works, it's got some files (meta.json), and a "senter" subdir.

dhoesele replied:
I did **not** run the code yet. 
I was just looking over the changes and saw the while True and in the except block, it just prints the exception and returns None. I expected some retry counter similar to the other functions in networking.py `def get(url: str, retry_count: int = 3)` or `def get_from(urls: list[str], retry_count: int = 3)`

lecoqnicolas replied:
The while True loop ensures the try branches goes on until the model is cached, no need for a retry. Exception will trigger on real exceptions only (Spacy repo error response), not if the download is simply delayed.

lecoqnicolas commented:
Actually, the comments prompted some code reviewing:
- renamed variables "xx_model_path" to "xx_pack" in package
- renamed variable spacy_cache to cahed_spacy in sbd
- swapped some lines and added comments to improve code coherence
- added logs for possible error explainability and follow-up.

lecoqnicolas commented:
Actually, I know how to make à cleaner PR. Save the code somewhere and rebase from master should do. I'll do this tomorrow to make the whole thing presentable.
